### DISCUSSION TITLE
An "--extend-fixable" option to extend on-save fixable rules with on-commit fixable rules

### DISCUSSION THREAD

**Discussion by lsorber** [astral-sh/ruff#4272] [Ideas]
Hi,

For the [Poetry Cookiecutter](https://github.com/radix-ai/poetry-cookiecutter) template that we maintain, we'd like to be able to distinguish between two types of fixable rules:

1. IDE fixable rules: fixes that you don't mind running on-save.
2. Commit-time (or later) fixable rules: a superset of the IDE fixable rules.

Examples of rules that you don't want to run on-save (1), but that may want to run on-commit (2) are:

- ERA001: remove commented out code (e.g. you comment out a section to run a test, but don't want these to be removed)
- F401: remove unused imports (e.g. you start a new file, type `import numpy as np`, then hit save and see the import removed)
- T201: remove print statements (e.g. you added print statement to debug something)

Is it correct that `--fixable` overrides (and does not extend) the fixable setting in `pyproject.toml`? Is there a way to extend `pyproject.toml`'s fixable setting with the CLI? Or would this be a new feature?

Thank you!

**Comment by charliermarsh** [comment]
I think it would make sense to support `--extend-fixable` and `--extend-unfixable`, to mirror `--select` and `--extend-select`. You're right that the current version overrides the `pyproject.toml`.

**Comment by charliermarsh** [comment]
Exists in the next release.
